src/networks/metrics.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019-2024 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::sync::Arc;
+
+use once_cell::sync::Lazy;
+use prometheus_client::metrics::gauge::Gauge;
+
+use super::ChainConfig;
+
+pub static EXPECTED_NETWORK_HEIGHT: Lazy<Gauge> = Lazy::new(|| {
+    let metric = Gauge::default();
+    crate::metrics::default_registry().register(
+        "expected_network_height",
+        "The expected network height based on the current time and the genesis block time.",
+        metric.clone(),
+    );
+    metric
+});
+
+/// Task to periodically update network-related metrics.
+pub async fn network_metrics_loop(config: Arc<ChainConfig>, genesis_timestamp: u64) {
```

Comment:
I think we could register a (`Collector`)[https://docs.rs/prometheus-client/latest/prometheus_client/collector/trait.Collector.html] instead of running a loop

Reply:
I didn't know about this, I'll look it up. Thanks.